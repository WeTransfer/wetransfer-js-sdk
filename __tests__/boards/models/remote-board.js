const { RemoteBoard } = require('../../../src/boards/models');

describe('RemoteBoard model', () => {
  let board = null;
  beforeEach(() => {
    board = new RemoteBoard({
      id: 'random-hash',
      name: 'Little kittens',
      description: 'Something about cats, most probably.',
      state: 'uploading',
      url: 'https://we.tl/s-random-hash',
      items: [],
    });
  });

  describe('contructor', () => {
    it('should create a normalized board', () => {
      expect(board).toMatchSnapshot();
    });

    it('should normalize provided items', () => {
      board = new RemoteBoard({
        id: 'random-hash',
        name: 'Little kittens',
        description: 'Something about cats, most probably.',
        state: 'uploading',
        url: 'https://we.tl/s-random-hash',
        items: [
          {
            id: 'random-hash',
            name: 'kittie.gif',
            type: 'file',
          },
          {
            id: 'random-hash',
            url: 'https://wetransfer.com',
            meta: {
              title: 'WeTransfer',
            },
            type: 'link',
          },
        ],
      });

      expect(board.items).toMatchSnapshot();
      expect(board.file).toMatchSnapshot();
      expect(board.links).toMatchSnapshot();
    });
  });

  describe('when items are added to the board', () => {
    beforeEach(() => {
      const files = [
        {
          id: 'random-hash',
          name: 'kittie.gif',
          type: 'file',
        },
      ];

      const links = [
        {
          id: 'random-hash',
          url: 'https://wetransfer.com',
          meta: {
            title: 'WeTransfer',
          },
          type: 'link',
        },
      ];

      board.addFiles(...files);
      board.addLinks(...links);
    });

    it('should add items to the board', () => {
      expect(board.items).toMatchSnapshot();
    });

    describe('files property', () => {
      it('should return only files', () => {
        expect(board.files).toMatchSnapshot();
      });
    });

    describe('links property', () => {
      it('should return only links', () => {
        expect(board.links).toMatchSnapshot();
      });
    });
  });
});
