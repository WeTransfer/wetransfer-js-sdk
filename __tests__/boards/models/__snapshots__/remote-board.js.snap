// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoteBoard model contructor should create a normalized board 1`] = `
RemoteBoard {
  "description": "Something about cats, most probably.",
  "id": "random-hash",
  "items": Array [],
  "name": "Little kittens",
  "state": "uploading",
  "url": "https://we.tl/s-random-hash",
}
`;

exports[`RemoteBoard model contructor should normalize provided items 1`] = `
Array [
  Object {
    "id": "random-hash",
    "multipart": undefined,
    "name": "kittie.gif",
    "size": undefined,
    "type": "file",
  },
  Object {
    "id": "random-hash",
    "meta": Object {
      "title": "WeTransfer",
    },
    "type": "link",
    "url": "https://wetransfer.com",
  },
]
`;

exports[`RemoteBoard model contructor should normalize provided items 2`] = `undefined`;

exports[`RemoteBoard model contructor should normalize provided items 3`] = `
Array [
  Object {
    "id": "random-hash",
    "meta": Object {
      "title": "WeTransfer",
    },
    "type": "link",
    "url": "https://wetransfer.com",
  },
]
`;

exports[`RemoteBoard model when items are added to the board files property should return only files 1`] = `
Array [
  Object {
    "id": "random-hash",
    "multipart": undefined,
    "name": "kittie.gif",
    "size": undefined,
    "type": "file",
  },
]
`;

exports[`RemoteBoard model when items are added to the board links property should return only links 1`] = `
Array [
  Object {
    "id": "random-hash",
    "meta": Object {
      "title": "WeTransfer",
    },
    "type": "link",
    "url": "https://wetransfer.com",
  },
]
`;

exports[`RemoteBoard model when items are added to the board should add items to the board 1`] = `
Array [
  Object {
    "id": "random-hash",
    "multipart": undefined,
    "name": "kittie.gif",
    "size": undefined,
    "type": "file",
  },
  Object {
    "id": "random-hash",
    "meta": Object {
      "title": "WeTransfer",
    },
    "type": "link",
    "url": "https://wetransfer.com",
  },
]
`;
